### PR TITLE
Lavaland fixes, continued

### PR DESCRIPTION
**Pull request by Markolie**
**What does this PR do:**
See the changelog.

Fixes https://github.com/ParadiseSS13/Paradise/issues/11651.
Fixes https://github.com/ParadiseSS13/Paradise/issues/11654.

🆑 Markolie
add: The Ancient Goliath now has a small chance of spawning instead of regular goliaths.
add: Survival capsules now have a NanoMed.
tweak: Water bottles now behave as glass containers.
tweak: The behaviour of glass reagent containers has changed. When used on non-mobs, contents will now only be spilled on harm intent. When used on mobs, on any intent aside from harm the contents will be fed to the mob like regular drinks. On harm intent it will still be spilled.
tweak: Hairless hide can now be made wet using most sources of water (minimum volume of 10), instead of just washing machines.
fix: Added some additional offstation role checks to ensure ghost spawner roles don't become antagonists, are kidnap targets or are counted towards station goals.
fix: Fixed some piping in the Lavaland Syndicate base incinerator and made sure the vault door starts locked.
fix: Fixed the Lavaland swarmers getting stuck on propulsion.
fix: Fix goliath tentacles sometimes being left behind.
fix: The walls around the Lavaland Syndicate base turbine have been coated, to prevent them from starting a plasma fire in the base.
fix: Syndicate agents can now access the air alarms on the Lavaland Syndicate base.
fix: Air alarm control computers can no longer access the air alarms on the Lavaland Syndicate base.
fix: Fixed the disk compartmentalizer/drying rack missing a sprite under certain circumstances.
fix: Fixed the Lavaland Syndicate base triggering atmospherics alerts.
fix: You can now properly place tiles on basalt.
/🆑

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/mining/goliath.dm [hunk 1]
```diff
@@ -74,44 +74,104 @@
...
+	icon_state = "Goliath"
+	icon_living = "Goliath"
+	icon_aggro = "Goliath_alert"
+	icon_dead = "Goliath_dead"
+	maxHealth = 400
+	health = 400
+	speed = 4
+	pre_attack_icon = "Goliath_preattack"
+	throw_message = "does nothing to the rocky hide of the"
+	loot = list(/obj/item/stack/sheet/animalhide/goliath_hide) //A throwback to the asteroid days
```
Comment (KasparoVy) [inline code/modules/mob/living/simple_animal/hostile/mining/goliath.dm]: Xenoarch accidentally killing mining by finding a SM shard 😂

**Comment by datlo** [comment]
Have you tested running the syndicate base turbine with the exhaust open? As lava turfs dont remove plasma/gases like space turfs do, I think youll start a plasmafire outside the base regardless of turbine coating.

**Comment by KasparoVy** [comment]
Flame on brother. What a hilarious place to put a turbine anyway

**Comment by Markolie** [comment]
@datlo Aye, the coated walls around the turbine will stop it from destroying the base. Because of planetary atmos it will eventually subside when the exhaust doors are closed.

File: code/modules/mining/mine_items.dm [hunk 2]
```diff
@@ -292,351 +292,3 @@
 		colorindex = 0
 	icon_state = "mobcap[colorindex]"
 	update_icon()
-
-/*****************************Survival Pod********************************/
```
Comment (Crazylemon64) [inline code/modules/mining/mine_items.dm]: Any changes made to this region?

File: code/modules/reagents/reagent_containers/glass_containers.dm [hunk 3]
```diff
@@ -117,11 +143,11 @@
 		return
 
 	else if(reagents.total_volume)
-		to_chat(user, "<span class='notice'>You splash the solution onto [target].</span>")
-		reagents.reaction(target, TOUCH)
-		spawn(5) reagents.clear_reagents()
-		return
-
+		if(user.a_intent == INTENT_HARM)
```
Comment (Crazylemon64) [inline code/modules/reagents/reagent_containers/glass_containers.dm]: This `if` clause seems like it should be bundled into the `else if` one line above

File: code/modules/mining/mine_items.dm (hunk 2 above)
Comment (Markolie) [reply code/modules/mining/mine_items.dm]: I added the NanoMed, switched the survival pod to be loaded by default with an empty subtype (to be consistent with /tg/), replaced the telecrystal artifact with a stack of twenty (which are fake anyway) and fixed some formatting. That also happens to fix a bug where you can't change the type of the smartfridge circuit board.

File: code/game/turfs/turf.dm [hunk 4]
```diff
@@ -180,11 +180,11 @@
 	if(L)
 		qdel(L)
 
-/turf/proc/TerraformTurf(path, defer_change = FALSE, keep_icon = TRUE, ignore_air = FALSE)
-	return ChangeTurf(path, defer_change, keep_icon, ignore_air)
+/turf/proc/TerraformTurf(path, defer_change = FALSE, ignore_air = FALSE)
+	return ChangeTurf(path, ignore_air)
```
Comment (Crazylemon64) [inline code/game/turfs/turf.dm]: It doesn't look like these arguments line up correctly - `ignore_air` here lines up with `defer_change` of `ChangeTurf`

File: code/game/turfs/simulated/floor.dm [hunk 5]
```diff
@@ -120,30 +120,31 @@ var/list/icons_to_ignore_at_floor_init = list("damaged1","damaged2","damaged3","
 /turf/simulated/floor/proc/make_plating()
 	return ChangeTurf(/turf/simulated/floor/plating)
 
-/turf/simulated/floor/ChangeTurf(turf/simulated/floor/T, defer_change = FALSE, keep_icon = TRUE)
-	if(!istype(src,/turf/simulated/floor)) return ..() //fucking turfs switch the fucking src of the fucking running procs
-	if(!ispath(T,/turf/simulated/floor)) return ..()
-	var/old_icon = icon_regular_floor
-	var/old_plating = icon_plating
-	var/old_dir = dir
+/turf/simulated/floor/ChangeTurf(turf/simulated/floor/T, defer_change = FALSE, ignore_air = FALSE)
```
Comment (Crazylemon64) [inline code/game/turfs/simulated/floor.dm]: You don't seem to change the way this function is called in the map loader,
`modules/awaymissions/maploader/reader.dm`.
That's the one piece of code that uses all 3 of these arguments. - `T.ChangeTurf(path, 1, 0)`
Are you sure the effects of this change will not adversely affect the maploader?

Comment (Markolie) [reply code/game/turfs/simulated/floor.dm]: I just ran another test: both ruins and away missions load totally fine (although I did forget to remove the second false parameter). /tg/ got rid of the keep_icons a while back, and I have no idea what we even used it for. My guess is something like icon smoothing, which is handled properly now anyway.

**Pull request merged**